Corzbank.Services/ExchangeService.cs

```diff
@@ -0,0 +1,109 @@
+﻿using Corzbank.Data.Entities;
+using Corzbank.Data.Enums;
+using Corzbank.Helpers;
+using Corzbank.Services.Interfaces;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Corzbank.Services
+{
+    public class ExchangeService : IExchangeService
+    {
+        private readonly GenericService<Exchange> _genericService;
+
+        public ExchangeService(GenericService<Exchange> genericService)
+        {
+            _genericService = genericService;
+        }
+        
+        public async Task<IEnumerable<Exchange>> GetValues()
+        {
+            var result = await _genericService.GetRange();
+
+            return result;
+        }
+
+        public async Task<Exchange> GetValueById(Guid id)
+        {
+            var result = await _genericService.Get(id);
+
+            return result;
+        }
+
+        public async Task<bool> CreateExchage()
+        {
+            dynamic valuesArray = GetValuesForExchange.GetValues();
+
+            if (!_genericService.CheckByCondition(e => e.ExchangeCurrency == Currency.EUR || e.ExchangeCurrency == Currency.USD))
+            {
+                foreach (var value in valuesArray)
```

Comment:
after you parse from helper, that loop won't be necessary

Reply:
resolved